src/proxy-ui-api/frontend/src/views/AppLogin.vue

```diff
@@ -175,6 +175,7 @@ export default (
           // Auth ok. Start phase 2 (fetch user data and current security server info).
           this.fetchUserData();
           this.fetchSecurityServerVersion();
```

Comment:
Also you need to make each of these functions to return the promise.... see there...
Additionally on DRY optimization would be to remove the catch -clauses with the same handling in the called functions and add it to here where these are called.
What about like this?
```suggestion
          Promise.all( [
          this.fetchUserData(),
          this.fetchSecurityServerVersion(),
          this.fetchSecurityServerNodeType()
          ]).catch(....) 
          
```

Reply:
This was a good find! 👍

All of the `fetch` functions are executed in parallel. When logging in to an uninitialized Security Server with a user that does not have permissions to init the server, the code will sometimes be able to fetch the version and node type info from backend before the `fetchUserData()` has finished and redirected the user to the login (i.e. logged out)

I will sort this out. I will probably refactor the whole file into using async-await and try-catch. At the moment it uses promise chains and sometimes `async` and sometimes also `await` so it makes it kinda hard to read.

Reply:
I refactored the logic in this file to use async-await and try-catch. It is now working correctly. 

I would advocate using async-await and try-catch everywhere instead of promise chains. Makes the code way cleaner and easier to read. Also, perhaps this sort of bugs could be easier to notice 🤔